src/H5MF.c

```diff
@@ -1058,7 +1058,6 @@ H5MF_xfree(H5F_t *f, H5FD_mem_t alloc_type, haddr_t addr, hsize_t size)
     assert(f);
     if (!H5_addr_defined(addr) || 0 == size)
         HGOTO_DONE(SUCCEED);
-    assert(addr != 0); /* Can't deallocate the superblock :-) */
```

Comment:
The intention here is clear, but it seems like there is a valid reason to call H5MF_xfree on the superblock for these changes. The alternative would be to create a counterpart function solely for the superblock, but that didn't really seem worth it.

Reply:
Put this back, as the call to `H5MF_xfree` for freeing the file space allocated for the superblock was occasionally allocating a new sector in the file free space sections and causing more issues. Just calling `H5MF_try_close` appears to be enough to fix the issues here.